From 50abe303df7a7278cde2a7d152e9fe5037d9010d Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Wed, 2 Apr 2014 20:44:43 +0200
Subject: [PATCH 116/118] BUILD: adjust makefile for AIX 5.1

AIX 5.1 has trouble with ss_family which is __ss_family there.
Just remap it in the makefile and provide a new target.
---
 Makefile | 10 +++++++++-
 README   |  1 +
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 881297b..d47e676 100644
--- a/Makefile
+++ b/Makefile
@@ -86,7 +86,7 @@ DOCDIR = $(PREFIX)/doc/haproxy
 # Use TARGET=<target_name> to optimize for a specifc target OS among the
 # following list (use the default "generic" if uncertain) :
 #    generic, linux22, linux24, linux24e, linux26, solaris,
-#    freebsd, openbsd, cygwin, custom, aix52
+#    freebsd, openbsd, cygwin, custom, aix51, aix52
 TARGET =
 
 #### TARGET CPU
@@ -285,6 +285,13 @@ ifeq ($(TARGET),openbsd)
   USE_KQUEUE     = implicit
   USE_TPROXY     = implicit
 else
+ifeq ($(TARGET),aix51)
+  # This is for AIX 5.1
+  USE_POLL        = implicit
+  USE_LIBCRYPT    = implicit
+  TARGET_CFLAGS   = -Dss_family=__ss_family
+  DEBUG_CFLAGS    =
+else
 ifeq ($(TARGET),aix52)
   # This is for AIX 5.2 and later
   USE_POLL        = implicit
@@ -300,6 +307,7 @@ ifeq ($(TARGET),cygwin)
   TARGET_CFLAGS  = $(if $(filter 1.5.%, $(shell uname -r)), -DUSE_IPV6 -DAF_INET6=23 -DINET6_ADDRSTRLEN=46, )
 endif # cygwin
 endif # aix52
+endif # aix51
 endif # openbsd
 endif # osx
 endif # freebsd
diff --git a/README b/README
index 0f3a1f9..d2413a8 100644
--- a/README
+++ b/README
@@ -30,6 +30,7 @@ and assign it to the TARGET variable :
   - freebsd     for FreeBSD 5 to 8.0 (others untested)
   - osx         for Mac OS/X
   - openbsd     for OpenBSD 3.1 to 5.2 (others untested)
+  - aix51       for AIX 5.1
   - aix52       for AIX 5.2
   - cygwin      for Cygwin
   - generic     for any other OS.
-- 
1.8.3.2

